src/DependencyInjection/Configuration.php

```diff
@@ -50,6 +40,31 @@ public function getConfigTreeBuilder()
 
             ->end();
 
+
         return $treeBuilder;
     }
+
+    private function addAuth0SdkConfiguration(ArrayNodeDefinition $rootNode): void
+    {
+        $sdkNode = $rootNode
+            ->children()
+                ->arrayNode('sdk')
+                    ->children();
+
+
+        $sdkConfigurationRefClass = new \ReflectionClass(SdkConfiguration::class);
+        $constructor = $sdkConfigurationRefClass->getConstructor();
+
+        foreach ($constructor->getParameters() as $parameter) {
+            if ('configuration' === $parameter->getName()) {
+                continue;
+            }
+
+            $defaultValue = $parameter->isOptional() ? $parameter->getDefaultValue() : null;
+
+            $sdkNode
+                ->scalarNode($parameter->getName())
```

Comment:
Lets use `variableNode()` if the parameter is an array.

Reply:
I made some more logic with reflection to type correctly all nodes